osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/application/mapping/PushSetupAlarmConverter.java

```diff
@@ -34,23 +40,48 @@
     if (source == null) {
       return null;
     }
-    if (!source.hasSendDestinationAndMethod()) {
-      throw new IllegalArgumentException(
-          "Unable to map PushSetupAlarm without SendDestinationAndMethod");
-    }
-    final SendDestinationAndMethod sendDestinationAndMethod = source.getSendDestinationAndMethod();
-    final String destination = sendDestinationAndMethod.getDestination();
-    if (!destination.matches("\\S++:\\d++")) {
-      throw new IllegalArgumentException(
-          "Unable to parse destination as \"<host>:<port>\": " + destination);
-    }
-    final String[] hostAndPort = destination.split(":");
+
     final org.opensmartgridplatform.adapter.ws.schema.smartmetering.configuration.PushSetupAlarm
         pushSetupAlarm =
             new org.opensmartgridplatform.adapter.ws.schema.smartmetering.configuration
                 .PushSetupAlarm();
-    pushSetupAlarm.setHost(hostAndPort[0]);
-    pushSetupAlarm.setPort(new BigInteger(hostAndPort[1]));
+
+    if (source.hasSendDestinationAndMethod()) {
+      final SendDestinationAndMethod sendDestinationAndMethod =
+          source.getSendDestinationAndMethod();
+      final String destination = sendDestinationAndMethod.getDestination();
+      if (!destination.matches("\\S++:\\d++")) {
+        throw new IllegalArgumentException(
+            "Unable to parse destination as \"<host>:<port>\": " + destination);
+      }
+      final String[] hostAndPort = destination.split(":");
+      pushSetupAlarm.setHost(hostAndPort[0]);
+      pushSetupAlarm.setPort(new BigInteger(hostAndPort[1]));
+    }
+
+    if (source.hasPushObjectList()) {
+      final List<CosemObjectDefinition> sourcePushObjectList = source.getPushObjectList();
+      for (final CosemObjectDefinition sourcePushObject : sourcePushObjectList) {
+
+        final PushObject convertedPushObject = new PushObject();
+
+        final ObisCodeValues convertedObisCode = new ObisCodeValues();
+        final CosemObisCode sourceObisCode = sourcePushObject.getLogicalName();
+        convertedObisCode.setA((short) sourceObisCode.getA());
+        convertedObisCode.setB((short) sourceObisCode.getB());
+        convertedObisCode.setC((short) sourceObisCode.getC());
+        convertedObisCode.setD((short) sourceObisCode.getD());
+        convertedObisCode.setE((short) sourceObisCode.getE());
+        convertedObisCode.setF((short) sourceObisCode.getF());
+
+        convertedPushObject.setClassId(sourcePushObject.getClassId());
+        convertedPushObject.setLogicalName(convertedObisCode);
+        convertedPushObject.setAttributeIndex((byte) sourcePushObject.getAttributeIndex());
+        convertedPushObject.setDataIndex(sourcePushObject.getDataIndex());
+        pushSetupAlarm.getPushObjectList().add(convertedPushObject);
```

Comment:
Looks like this could be refactored using (new) Orika converters for mapping:
`CosemObisCode` <---> `ObisCodeValues`
`CosemObjectDefinition` <---> `PushObject`

Which, I think, then might result in something like:
```
   List<PushObject> convertedPushObjectList = this.mapperFacade.mapAsList(source.getObjectList(), PushObject.class))
   pushSetupAlarm.getPushObjectList().addAll(convertedPushObjectList);
```
Something similar applies to the `convertFrom` method as well.

Reply:
Added converters